SimPEG/potential_fields/gravity/simulation.py

```diff
@@ -95,8 +251,10 @@ def G(self):
         Gravity forward operator
         """
         if getattr(self, "_G", None) is None:
-            self._G = self.linear_operator()
-
+            if self.engine == "choclo":
+                self._G = self._sensitivity_matrix()
```

Comment:
**suggestion**: Same here. The issue with new attribute names is that it makes grav differ a lot from the base class (and sub-classes). I would vote for `engine` to be on the base class, such that we can eventually roll out different engines on the subclasses.

Reply:
As I replied before to @jedman, I didn't want to touch the parent class too much here. I was focusing on implementing the forward for gravity only. If I would have added the `engine` attribute to the parent class in this PR, I would also had to modify the behaviour of all child classes. For example, the mag simulation should throw an error if the user is trying to pass `engine="choclo"` (at least for now). And I didn't want to do that at this stage.

I think once we are happy with the design of the class and how we expect users to interact with it, we can move stuff like this one to the parent class. Specially when implementing the mag simulation using Choclo, where the two child classes can reuse some methods.

Does it make sense?